### PR TITLE
Added GitHub workflows for CI and PyPi publish

### PR DESCRIPTION
**Pull request by mraarif**
Relevant JIRA issue: https://openedx.atlassian.net/browse/BOM-2102

### CODE REVIEW

**Comment by mraarif** [comment]
repo needs to update its codecov configuration to use report uploaded by GH workflows instead of travis.

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  push:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.5
```
Comment (jmbowman) [inline .github/workflows/pypi-publish.yml:18]: We should probably switch this to 3.8.

File: .github/workflows/pypi-publish.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Install pip
+        run: pip install -r requirements/pip.txt
+
+      - name: Build package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish to PyPi
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: edx
```
Comment (jmbowman) [inline .github/workflows/pypi-publish.yml:29]: I thought we're supposed to use `__token__` as the user when using a PyPI API token? https://github.com/pypa/gh-action-pypi-publish#usage

File: Makefile [hunk 3]
```diff
@@ -64,12 +64,13 @@ upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -qr requirements/pip-tools.txt
 	# Make sure to compile files after any other files they include!
+	$(PIP_COMPILE) --allow-unsafe --rebuild -o requirements/pip.txt requirements/pip.in
 	$(PIP_COMPILE) --rebuild -o requirements/pip-tools.txt requirements/pip-tools.in
```
Comment (jmbowman) [inline Makefile:68]: Shouldn't need `--rebuild` on this line anymore since it's now on the previous one.

File: requirements/dev.in [hunk 4]
```diff
@@ -4,7 +4,7 @@
 -r pip-tools.txt          # pip-tools and its dependencies, for managing requirements files
 -r quality.txt            # Dependencies for quality check utilities
 -r test.txt               # Dependencies for running unit tests
--r travis.txt             # Dependencies for running tox
+-r ci.txt             # Dependencies for running tox
```
Comment (jmbowman) [inline requirements/dev.in:7]: Nit: would be nice to keep the comment aligned with the ones on previous lines

File: tox.ini [hunk 5]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py35-django22-celery{44}-drf39,py38-django{22,30}-celery{44}-drf{39,310,311,312}
+envlist = py{3.5}-django22-celery{44}-drf39,py{3.8,3.9}-django{22,30,31}-celery{44}-drf{39,310,311,312},quality,docs
```
Comment (jmbowman) [inline tox.ini:2]: Again, can we just drop the 3.5 testing now?

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+name: Python CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (jmbowman) [inline .github/workflows/ci.yml:7]: Ned pointed out today that we probably want to run tests on PRs against any branch, discussion in https://edx-internal.slack.com/archives/CD89W7Z7F/p1606849364465500

File: tox.ini [hunk 7]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py35-django22-celery{44}-drf39,py38-django{22,30}-celery{44}-drf{39,310,311,312}
+envlist = py{3.8,3.9}-django{22,30,31}-celery{44}-drf{39,310,311,312},quality,docs
```
Comment (jmbowman) [inline tox.ini:2]: At first I'd assumed the `38` -> `3.8` switch was needed to simplify the GitHub Action, but it doesn't look that way.  Was there a specific need for that change?  The Travis test config still says `py38`, this isn't causing any trouble there?

Comment (mraarif) [reply tox.ini]: you're right, that was my initial understanding, turned out to be a bit more intuitive than I first thought.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**